File: 41-first-missing-positive/first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        HashSet<Integer> st = new HashSet<>();

        for(int i = 0;i<nums.length;i++){
                if(nums[i]>0){
                    st.add(nums[i]);
                }
        }

        int missing = 1;
        for(int j : st){
            if(st.contains(missing)){
                missing++;
            }
        }
        return missing;
    }
}

File: 41-first-missing-positive/first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int n=nums.size();
        vector<bool> s(n+1,false);

        for(int no: nums){
            if(no>0 && no<=n ){
                s[no]=true;
            }    
        }
        for(int i=1;i<=n;i++){
            if(!s[i]){
                return i;
            }
        }
        return n+1;
    }
};